include/odp/api/spec/timer.h

```diff
@@ -37,15 +37,32 @@ extern "C" {
 
 /**
  * Clock sources for timers in timer pool.
+ *
+ * ODP_CLOCK_CPU is the default clock source and it is supported always. It is implementation
+ * defined which other clock sources are supported. See implementation documentation for
+ * possible clock sources and their enumeration to these values.
  */
 typedef enum {
-	/** Use CPU clock as clock source for timers */
+	/** CPU clock as clock source for timers (default) */
 	ODP_CLOCK_CPU,
-	/** Use external clock as clock source for timers */
-	ODP_CLOCK_EXT
-	/* Platform dependent which other clock sources exist */
+
```

Comment:
Timer API functions have this enumeration as input parameter. It is backwards compatible to add new  enumerations vs e.g. introduce a new parameter (int ext_clock_index).

Implementation manual could tell that e.g. EXT_0 is GPIO X, EXT_1 is driven by PTP clock, etc. We can add (now or) later additional clock sources per type if that is needed (ODP_CLOCK_GPIO_0, ODP_CLOCK_PTP, ...). But even then these EXT_0 ... _3 could remain.

Clock source enumeration needs some fixed format, so that validation test can check also those.